lib/line-index/src/lib.rs

```diff
@@ -115,29 +125,34 @@ impl LineIndex {
         LineIndex { newlines, line_wide_chars }
     }
 
+    /// Transforms the `TextSize` into a `LineCol`.
     pub fn line_col(&self, offset: TextSize) -> LineCol {
         let line = self.newlines.partition_point(|&it| it <= offset) - 1;
         let line_start_offset = self.newlines[line];
         let col = offset - line_start_offset;
         LineCol { line: line as u32, col: col.into() }
     }
 
+    /// Transforms the `LineCol` into a `TextSize`.
     pub fn offset(&self, line_col: LineCol) -> Option<TextSize> {
         self.newlines
             .get(line_col.line as usize)
             .map(|offset| offset + TextSize::from(line_col.col))
     }
 
+    /// Transforms the `LineCol` with the given `WideEncoding` into a `WideLineCol`.
     pub fn to_wide(&self, enc: WideEncoding, line_col: LineCol) -> WideLineCol {
         let col = self.utf8_to_wide_col(enc, line_col.line, line_col.col.into());
         WideLineCol { line: line_col.line, col: col as u32 }
     }
 
+    /// Transforms the `WideLineCol` with the given `WideEncoding` into a `LineCol`.
     pub fn to_utf8(&self, enc: WideEncoding, line_col: WideLineCol) -> LineCol {
         let col = self.wide_to_utf8_col(enc, line_col.line, line_col.col);
         LineCol { line: line_col.line, col: col.into() }
     }
 
+    /// Returns an iterator over the ranges for the lines.
     pub fn lines(&self, range: TextRange) -> impl Iterator<Item = TextRange> + '_ {
```

Comment:
It's unclear which lines are returnd, lines intersecting range, or lines contained in the range.

Reply:
from my reading of the function, it looks like it's the non empty ranges given by the sequence

$$
(R_\textsf{start}, x_1), (x_1, x_2), \dots, (x_{n-1}, x_n), (x_n, R_\textsf{end})
$$

where $R$ is the input range and each $x_i$ is the position of a newline.

how would you characterize that? (maybe the above pseudocode is sufficient?)

Reply:
i added basically this^ to the docs